modules/ob-securely-referencing-string-parameters.adoc

```diff
@@ -0,0 +1,83 @@
+// This module is included in the following assembly:
+//
+// * builds/configuring-build-strategies.adoc
+
+:_content-type: REFERENCE
+[id="ob-securely-referencing-string-parameters_{context}"]
+= Secure referencing of string parameters
+
+String parameters are used when you define environment variables, arguments, or images in a `BuildStrategy` or `ClusterBuildStrategy` custom resource (CR). In your build strategy steps, you can reference string parameters by using the `$(params.your-parameter-name)` syntax.
+
+[NOTE]
+====
+You can also reference system parameters and strategy parameters by using the `$(params.your-parameter-name)` syntax in your build strategy steps.
+====
+
+In the pod, all `$(params.your-parameter-name)` variables are replaced by actual strings. However, you must pay attention when you reference a string parameter in an argument by using an inline script. For example, to securely pass a parameter value into an argument defined with a script, you can choose one of the following approaches:
+
+* Use environment variables
+* Use arguments
+
+[id="using-environment-variables-approach_{context}"]
+== Using environment variables approach
```

Comment:
"Using" makes this and the other section seem like procedures. 

Could reduce repetition of "approach" by removing H2s and turning them into labeled code blocks.

Ex:
```
.A BuildStrategy CR that passes values by using environment variables
[source,yaml]
----
apiVersion: shipwright.io/v1beta1
kind: BuildStrategy
metadata:
  name: buildpacks-v3
spec:
  parameters:
    - name: sample-parameter
      description: A sample parameter
      type: string
  steps:
    - name: sample-step
      env:
        - name: PARAM_SAMPLE_PARAMETER
          value: $(params.sample-parameter)
      command:
        - /bin/bash
      args:
        - -c
        - |
          set -euo pipefail
          some-tool --sample-argument "${PARAM_SAMPLE_PARAMETER}"
----
```